lib/easybill_rest_client/api_client.rb

```diff
@@ -26,55 +30,52 @@ def request_collection(method, endpoint, params = {})
     end
 
     def request(method, endpoint, params = {})
-      Retryable.retryable(
-        tries: tries,
-        sleep: retry_cool_off_time,
-        on: EasybillRestClient::TooManyRequests
-      ) do
-        response = perform_request(method, endpoint, params)
-        process_response(response)
+      @request_logger = RequestLogger.new(logger: logger, request_id: SecureRandom.hex(3))
```

Comment:
This makes `#request` not thread-safe. It probably doesn't matter in production, but it's bad design IMHO.